Update Documentation - Clearer client alternative usage and highlight the require use of error handling (#318)
Unleash/unleash-client-node: README.md

  const variant = unleash.getVariant('app.ToggleY');
});
```
Be aware that the `initialize` function will configure a _global_ Unleash instance. If you call this method multiple times the global instance will be changed. 

> #### <u>Constructing the Unleash client directly</u>

Reviewer: Is the `u` tag intended to underline the heading? It doesn't seem to work on GitHub (at least not in the preview), and I think we should leave the header styling up to the markdown renderer anyway 💁🏼

```suggestion
#### Constructing the Unleash client directly
```

> If you prefer to construct the Unleash instance yourself rather than using the global `initialize` method you can do so e.g.:

Reviewer: Nice! When would you want to do this / what problems does it solve? I think giving the users an example of when they'd want to do this would be useful 🤓

Also, a slight rewording suggestion:
```suggestion
You can also construct the Unleash instance yourself instead of via the `initialize` method. 
This lets you [...] // <- why is this useful?
```

Author: Yeah the idea was to stylize it so it looked like a proper subsection of section `2.1`, but not that important anyways 👍🏽 
My markdown renderer locally did render the underline properly so  i thought it would work on Github too 🤷🏽

```js
const { Unleash } = require('unleash-client');

const unleash = new Unleash({
  appName: 'my-app-name',
  url: 'http://unleash.herokuapp.com',
  customHeaders: {
    Authorization: 'API token',
  },
});

unleash.on('ready', console.log.bind(console, 'ready'));

// required error handling when using unleash directly  
unleash.on('error', console.error);
```

> **_IMPORTANT:_** When using unleash directly, error handling is required by adding an event listener for the error event. Otherwise your application may either not start up, or crash if its running when the application can't reach the unleash server.

Reviewer: This is great! But I think it might make sense to move it to before the code sample, so that it's:
1. introductory paragraph
2. error handling
3. code sample

What do you think? ☺️ 

Also a minor rewording suggestion:

```suggestion
**Important:** When Using unleash directly, you **must** handle errors yourself. This can be done by attaching an event listener to the `error` event, as shown in the example below. If you don't do this, your application may crash either on startup or at runtime if it can't reach the Unleash server or if it encounters other errors.
```

Author: > This is great! But I think it might make sense to move it to before the code sample, so that it's:
> 
> 1. introductory paragraph
> 2. error handling
> 3. code sample
> 
> What do you think? ☺️
I agree will change the order of things 👍🏽

Author: > 
> Also a minor rewording suggestion:
Thanks for the rewording suggestion, gave this text a couple of rounds but thought that it lacked something that's what PR's for 🙂  it reads much better now 👍🏽
